pkg/diagnostics/network/objects.go

```diff
@@ -0,0 +1,142 @@
+package network
+
+// Resource objects used by network diagnostics
+import (
+	"fmt"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclientcmd "k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	"k8s.io/kubernetes/pkg/util/intstr"
+
+	"github.com/openshift/origin/pkg/diagnostics/networkpod/util"
+)
+
+const (
+	busyboxImage               = "docker.io/busybox"
+	networkDiagTestPodSelector = "network-diag-pod-name"
+
+	testPodImage   = "docker.io/openshift/hello-openshift"
+	testPodPort    = 9876
+	testTargetPort = 8080
+)
+
+func GetNetworkDiagnosticsPod(command []string, podName, nodeName string) *kapi.Pod {
+	privileged := true
+	hostRootVolName := "host-root-dir"
+	secretVolName := "kconfig-secret"
+	secretDirBaseName := "secrets"
+	gracePeriod := int64(0)
+
+	pod := &kapi.Pod{
+		ObjectMeta: kapi.ObjectMeta{Name: podName},
+		Spec: kapi.PodSpec{
+			RestartPolicy:                 kapi.RestartPolicyNever,
+			TerminationGracePeriodSeconds: &gracePeriod,
+			ServiceAccountName:            util.NetworkDiagServiceAccountName,
+			SecurityContext: &kapi.PodSecurityContext{
+				HostPID:     true,
+				HostIPC:     true,
+				HostNetwork: true,
+			},
+			NodeName: nodeName,
+			Containers: []kapi.Container{
+				{
+					Name:            podName,
+					Image:           busyboxImage,
+					ImagePullPolicy: kapi.PullIfNotPresent,
+					SecurityContext: &kapi.SecurityContext{
+						Privileged: &privileged,
+						Capabilities: &kapi.Capabilities{
+							Add: []kapi.Capability{
+								// To run ping inside a container
+								"NET_ADMIN",
```

Comment:
pretty sure you don't need to set capabilities if you're privileged (in which case you also don't need a special new SCC)